bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -267,4 +276,55 @@ public void filter(@Nullable ContainerRequestContext requestContext) throws IOEx
             }
         }
     }
+
+    private boolean isImplicitUserRole(ContainerRequestContext requestContext) {
+        if (implicitUserRole) {
+            return true;
+        }
+        if (trustedNetworks.isEmpty()) {
+            return false;
+        }
+        return trustedNetworks.stream().anyMatch((networkCDIR) -> {
+            var clientIp = getClientIp(requestContext);
+            var isInTrustedNetwork = isIpInSubnet(clientIp, networkCDIR);
+            if (isInTrustedNetwork) {
+                logger.debug("Granted implicit user role to request with ip {} as part of network {}", clientIp,
+                        networkCDIR);
+            }
+            return isInTrustedNetwork;
+        });
```

Comment:
```suggestion
        String clientIp = getClientIp(requestContext);
        return trustedNetworks.stream().anyMatch(networkCIDR -> isInSubNet(clientIp, networkCIDR);
```

You could also create `SubnetUtils.SubnetInfo`s when parsing the config, so you can call `SubnetInfo::isInRange` here.

Reply:
Changes applied, thank you for the suggestion!
Package for SubnetUtils is not installed, that's why I look up for a function that relies on InetAddress.
Please reopen this if you think it's better to install the package and rely on SubnetUtils.